app/webpacker/components/RegistrationsV2/RegistrationEdit/Refunds.jsx

```diff
@@ -37,61 +37,77 @@ export default function Refunds({
-      onExit();
+      onSuccess();
     },
   });
-  return refundsLoading ? (
-    <Loading />
-  ) : (
-    !refundError && (
-      <Modal open={open} dimmer="blurring">
-        <Modal.Header>Available Refunds:</Modal.Header>
-        <Modal.Content>
-          <Table>
-            <Table.Header>
-              <Table.Header>Amount</Table.Header>
-              <Table.Header> </Table.Header>
-            </Table.Header>
-            <Table.Body>
-              {refunds.charges.map((refund) => (
-                <Table.Row key={refund.payment_id}>
-                  <Table.Cell>
-                    <Input
-                      labelPosition="right"
-                      type="text"
-                      placeholder={refund.amount}
-                    >
-                      <Label basic>$</Label>
-                      <input
-                        value={refundAmount}
-                        max={refund.amount}
-                        onChange={(event) => setRefundAmount(event.target.value)}
-                      />
-                    </Input>
-                  </Table.Cell>
-                  <Table.Cell>
-                    <Button
-                      onClick={() => refundMutation({
-                        competitionId,
-                        userId,
-                        paymentId: refund.payment_id,
-                        amount: refundAmount,
-                      })}
-                    >
-                      Refund Amount
-                    </Button>
-                  </Table.Cell>
-                </Table.Row>
-              ))}
-            </Table.Body>
-          </Table>
-        </Modal.Content>
-        <Modal.Actions>
-          <Button disabled={isMutating} onClick={onExit}>
-            Go Back
-          </Button>
-        </Modal.Actions>
-      </Modal>
-    )
+
+  if (refundsLoading) {
+    return <Loading />;
+  }
+
+  if (refunds.length === 0) {
+    return <Message success>All charges have been refunded</Message>;
+  }
+
+  return (
+    <>
+      <Header>Available Refunds:</Header>
+      <Table>
+        <Table.Header>
+          <Table.Row>
+            <Table.HeaderCell>Full Amount</Table.HeaderCell>
+            <Table.HeaderCell>Refund Amount </Table.HeaderCell>
+            <Table.HeaderCell />
+          </Table.Row>
+        </Table.Header>
+        <Table.Body>
+          {refunds.map((refund) => (
+            <RefundRow
+              refund={refund}
+              refundMutation={refundMutation}
+              isMutating={isMutating}
+              userId={userId}
+              competitionId={competitionId}
+              key={refund.payment_id}
+            />
+          ))}
+        </Table.Body>
+      </Table>
+    </>
+  );
+}
+
+function RefundRow({
+  refund, refundMutation, isMutating, userId, competitionId,
+}) {
+  const [amountToRefund, setAmountToRefund] = useState(refund.ruby_amount);
+
+  return (
+    <Table.Row key={refund.payment_id}>
```

Comment:
You don't need the key here, only where `RefundRow` is used.
```suggestion
    <Table.Row>
```